test/contracts/ContractInterface.h

```diff
@@ -62,15 +62,16 @@ class ContractInterface
 		BOOST_CHECK(call(_name + "(string,bytes32)", u256(0x40), _arg2, _arg1.length(), _arg1).empty());
 	}
 
-	u160 callStringReturnsAddress(std::string const& _name, std::string const& _arg)
+	util::h160 callStringReturnsAddress(std::string const& _name, std::string const& _arg)
 	{
 		bytes const& ret = call(_name + "(string)", u256(0x20), _arg.length(), _arg);
 		BOOST_REQUIRE(ret.size() == 0x20);
 		BOOST_CHECK(std::count(ret.begin(), ret.begin() + 12, 0) == 12);
+		// TODO: simplify this
 		return u160(u256(util::h256(ret)));
```

Comment:
Not fully sure what the intent here was.

Reply:
So this equals to `h160(u160(u256(h256(ret)))` which means taking the right aligned 160 bits of the result.